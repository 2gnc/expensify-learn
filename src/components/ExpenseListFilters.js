import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({ startDate, endDate }) => {
        this.props.onSetStartDate(startDate);
        this.props.onSetEndDate(endDate)
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    };
    onTextChange = (e) => {
        this.props.onSetTextFilter(e.target.value)
    };

    render() {
        return (
            <div>
            <input
                type="text"
                value={this.props.filters.text}
                onChange={this.onTextChange}
            />
            <select
                value={this.props.filters.sortBy}
                onChange={(e) => {
                if (e.target.value === 'date') {
                    this.props.onSortByDate()
                } else if (e.target.value === 'amount') {
                    this.props.onSortByAmount()
                }
                }}
            >
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
            <DateRangePicker
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false}
            />
            </div>
        );
    }
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    onSetStartDate: (data) => dispatch(setStartDate(data)),
    onSetEndDate: (data) => dispatch(setEndDate(data)),
    onSetTextFilter: (data) => dispatch(setTextFilter(data)),
    onSortByDate: () => dispatch(sortByDate()),
    onSortByAmount: () => dispatch(sortByAmount())
});
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
