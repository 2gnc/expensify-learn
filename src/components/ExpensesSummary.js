import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpenseSummary = ({count, total}) => {
    return (
        <div>
            <h2>Total: {count}</h2>
            <span>{numeral(total/100).format('$0,0.00')}</span>
        </div>
    )
};

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters);
    return {
        total: expensesTotal(expenses),
        count: expenses.length
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
