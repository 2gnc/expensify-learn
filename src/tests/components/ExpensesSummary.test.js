import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

describe('ExpenseSummary component', () => {
    it('renders $0.00 correctly if there is no expenses shown', () => {
        const wrapper = shallow(<ExpenseSummary count={0} total={0} />);
        const text = wrapper.find('span').text();
        expect(text).toBe('$0.00');
        expect(wrapper).toMatchSnapshot();
    }); 

    it('renders with one expense', () => {
        const wrapper = shallow(<ExpenseSummary count={1} total={12300} />);
        const text = wrapper.find('span').text();
        expect(text).toBe('$123.00');
        expect(wrapper).toMatchSnapshot();
    }); 
});
