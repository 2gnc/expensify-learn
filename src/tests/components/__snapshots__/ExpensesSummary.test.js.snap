// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseSummary component renders $0.00 correctly if there is no expenses shown 1`] = `
<div>
  <h2>
    Total: 
    0
  </h2>
  <span>
    $0.00
  </span>
</div>
`;

exports[`ExpenseSummary component renders with one expense 1`] = `
<div>
  <h2>
    Total: 
    1
  </h2>
  <span>
    $123.00
  </span>
</div>
`;
